src/ngraph/ngraph_nnvm_utils.h

```diff
@@ -74,6 +74,24 @@ inline TensorViewVector make_ngraph_placeholders(
   return out;
 }
 
+// creates and returns vector of TensorViews for corresponding NDArrays
+// reuses NDArray memory for each TensorView if req is not kAddTo
+inline TensorViewVector get_tensor_views(
+    const std::vector<mxnet::NDArray>& ndarrays,
+    std::shared_ptr<ngraph::runtime::Backend> backend,
+    const std::vector<mxnet::OpReqType>* req = nullptr) {
+  TensorViewVector out;
+  for (size_t i = 0; i < ndarrays.size(); ++i) {
+    auto shape = TShape_to_NShape(ndarrays[i].shape());
+    const auto& element_type = getType(ndarrays[i].dtype());
+    if ((req != nullptr) && ((*req)[i] == mxnet::kAddTo))
+      out.push_back(backend->make_primary_tensor_view(element_type, shape));
+    else
+      out.push_back(backend->make_primary_tensor_view(
+          element_type, shape, ndarrays[i].storage_handle().dptr));
```

Comment:
Does this API work for Argon and GPU?

Reply:
ngraph plans to support this api for all backends, in one form or other. GPU could technically re-use CUDA mem , but NDArray integration may not be as straightforward.

Until we have them fully supported/stabilized, I will modify PR to do CPU backend for now.

Reply:
@csullivan can you please review these changes. MXNet has its own cuda mem management, and this PR should pass that cuda mem to ngraph. Is there a simple model for GPU test

Reply:
Yep, you can run examples/image_recognition/train_mnist.py when compiling with GPU support and setting MXNET_NGRAPH_GPU=TRUE